File: Easy/Shortest Distance to a Character.py
class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
        n = len(s)
        result = [0] * n
        
        # set lastOccur to n so the first pass creates out-index values to the position that are not valid
        lastOccur = n      
        for i in range(n):
            if s[i] == c:
                lastOccur = i
            result[i] = abs(i - lastOccur)
        
        # set lastOccur to n so the first pass creates out-index values to the position that are not valid
        lastOccur = -1
        for i in range(n-1,-1,-1):
            if s[i] == c:
                lastOccur = i
            result[i] = min(result[i],abs(i - lastOccur))
        
        return result

File: LeetCode/C#/Easy/Shortest Distance to a Character.cs
public class Solution {
    public int[] ShortestToChar(string s, char c) {
        var locations = new int[s.Length];
        Array.Fill(locations, 0);
        
        int lastOccur = int.MinValue/2;
        for(int i=s.Length-1; i>=0; i--) {
            if(s[i] == c) {
                lastOccur = i;
            }
            locations[i] = Math.Abs(i - lastOccur);
        }

        lastOccur = int.MaxValue/2;
        for(int i=0; i < s.Length; i++) {
            if(s[i] == c) {
                lastOccur = i;
            }
            locations[i] = Math.Min(locations[i],  Math.Abs(i-lastOccur));
        }
        
        return locations;
    }
}